src/bin/hms-mqtt-publish/main.rs

```diff
@@ -83,7 +84,30 @@ fn main() {
         output_channels.push(Box::new(SimpleMqtt::<RumqttcWrapper>::new(&config)));
     }
 
+    if config.smiles_cooperation {
+        info!("S-Miles cloud cooperative mode enabled");
+    }
+
     loop {
+        // Do not query the inverter when the S-Miles cloud is about to update
+        if config.smiles_cooperation {
+            let now = SystemTime::now();
+            let duration_since_epoch = now.duration_since(UNIX_EPOCH).unwrap();
+            let seconds_since_epoch = duration_since_epoch.as_secs();
+
+            let seconds_in_current_minute = seconds_since_epoch % 60;
+            let minutes_since_epoch = seconds_since_epoch / 60;
+            let minutes_in_current_hour = minutes_since_epoch % 60;
+
+            // This is the time at which the S-Miles update seems to take place
+            // Adding some extra time before and after, in which we dont publish
+            if minutes_in_current_hour % 15 == 14 {
+                thread::sleep(Duration::from_millis(
+                    (15 + 60 - seconds_in_current_minute) * 1000,
```

Comment:
Do you know what the minimum sleep requirement is? Would it make sense to just sleep for a minute?